cmd/prometheus/main.go

```diff
@@ -287,6 +287,20 @@ func (c *flagConfig) setFeatureListOptions(logger *slog.Logger) error {
 	return nil
 }
 
+// parseCompressionType parses the two compression-related configuration values and returns the CompressionType. If
+// compression is enabled but the compressType is unrecognized, we default to Snappy compression.
+func parseCompressionType(compress bool, compressType compression.Type) (compression.Type, error) {
+	if compress {
+		if compressType == compression.None {
+			return compression.None, errors.New("got --storage.tsdb.wal-compression=true, " +
+				"but --storage.*.wal-compression-type was set to None; " +
+				"set --storage.tsdb.wal-compression=false and remove --storage.*.wal-compression-type for no compression")
+		}
+		return compressType, nil
+	}
+	return compression.None, nil
+}
```

Comment:
Reading this now, I understand that allowing passing compression None was not an accident. You'd like to handle it here in the following step.

I feel like we don't need this code, though. We can just let the flag lib handle which values are allowed, and we never need to worry about people passing None.

Reply:
Fair, unless we try to deprecate compress bool flag one day - but then we can add none back! Will fix.